tests/explicit-nulls/neg/notnull/J.java

```diff
@@ -0,0 +1,13 @@
+import java.util.*;
+import notnull.NotNull;
+
+public class J {
+
+    @NotNull
+    // TODO: remove annotaion after #7483
```

Comment:
Let's modify this test so that it continues to work even after the referenced bug is fixed.
We could do this by assigning `k` the result of a method, instead of a constant.

```Scala
String fooMethod() { return "hello"; }

@NotNull
public static final String k = fooMethod();
```

This way, `k` should never get a `ConstantType`.